### ISSUE TITLE
CA2208 does not consider lambda arguments as valid candidates

### ISSUE THREAD

**Issue by onyxmaster** [dotnet/roslyn-analyzers#1489]
#### Analyzer package

Microsoft.NetCore.Analyzers, version [2.6.0-beta2](https://www.nuget.org/packages/Microsoft.NetCore.Analyzers/2.6.0-beta2)

#### Analyzer

InstantiateArgumentExceptionsCorrectly

#### Repro steps
```
        void F()
        {
            Action<int> a =
                v =>
                {
                    throw new ArgumentException("", nameof(v));
                };
        }
```

#### Expected behavior
No warnings are reported.

#### Actual behavior
CA2208 is reported on ArgumentException instantiation.

**Comment by froufrou121** [comment]
I have the same issue with version 2.6.0.6241303. It would be nice if that was fixed rapidly.

**Comment by jamesqo** [comment]
@mavasani I have a PR implementing the fixer for this. Should I fix this bug there or do it in a separate PR?

**Comment by jinujoseph** [comment]
@jamesqo prefer separate PR so we can get the fix early and not blocked by fixer changes.

**Comment by mavasani** [comment]
Unable to repro this on latest FxCop analyzers package.
